### PR TITLE
feat: Execute charge price points settlement report

### PR DESCRIPTION
**Pull request by johevemi**
# Description

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py [hunk 1]
```diff
@@ -66,6 +67,8 @@ class Charge:
     charge_type: ChargeTypeDataProductValue
     charge_owner_id: str
     is_tax: bool
+    charge_price: Decimal
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py:70]: I donøt think this should be part of Charge? The purpose of Charge is to have ChargePriceInformation. So, the name is bad. In your case you could have a constant CHARGE_PRICE and CHARGE_TIME?

File: source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py [hunk 2]
```diff
@@ -157,6 +160,31 @@ def create_charge_link_periods(spark: SparkSession) -> DataFrame:
     return charge_link_periods_factory.create(spark, rows)
 
 
+def create_charge_price_points(spark: SparkSession) -> DataFrame:
+    """
+    Creates a DataFrame with charge prices data for testing purposes.
+    """
+
+    rows = []
+    for metering_point in _get_all_metering_points():
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py:169]: why loop over metering point?

File: source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py [hunk 3]
```diff
@@ -354,12 +382,16 @@ def _get_all_charges() -> list[Charge]:
...
+            charge_price=Decimal("10"),
+            charge_time=FROM_DATE,
         ),
         Charge(
             charge_key=f"4001_{ChargeTypeDataProductValue.TARIFF.value}_5790001330553",
             charge_code="4001",
             charge_type=ChargeTypeDataProductValue.TARIFF,
             charge_owner_id=CHARGE_OWNER_ID_WITH_TAX,
             is_tax=True,
+            charge_price=Decimal("12"),
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py:394]: should not be part of Charge

File: source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py]: removed

File: source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py:70]: ok

File: source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py (hunk 3 above)
Comment (johevemi) [reply source/databricks/settlement_report/tests/data_seeding/standard_wholesale_fixing_scenario_data_generator.py:393]: removed

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**